Support hasql 1.9

kokobd opened the pull request:
Closes #370

TeofilC commented on tests/Main.hs:
```diff
@@ -1,5 +1,6 @@
 {-# language BangPatterns #-}
 {-# language BlockArguments #-}
+{-# LANGUAGE CPP #-}
```
```suggestion
{-# language CPP #-}
```

TeofilC reviewed (changes requested):
Thanks! Have you tested this works correctly?

Could you also add a changelog entry using https://scriv.readthedocs.io/en/latest/

And make the language pragma consistent with the others.

I think otherwise this should be good to merge.

kokobd commented:
1. I tested in these environments, each with a fixed hasql constraint in cabal.project of `1.8.1.4` / `1.9.1.2`:
    - GHCup provided GHC 9.8.4 and 9.6.7, on `aarch64-darwin`
    - `nix build .#checks.aarch64-darwin.tests`
    - `nix build .#checks.x86_64-linux.tests`
2. The changelog is added and the language pragma issue is fixed.
3. I also made some nix related updates (these are not mentioned in changelog since they won't affect the released package):
    - I run `nix flake update` because the existing `flake.lock` was updated 22 months ago, which makes it impossible for nix users to test against hasql 1.9.
    - In `flake.nix` I changed it to use `eachDefaultSystem` so Mac users can enter the development shell. CI won't be able to run `nix flake check` because `nix flake check` wants to run the tests of all architectures on the same machine. I updated the CI to run `nix build .#checks.x86_64-linux.tests` instead.

TeofilC reviewed (approved):
Thanks!

kokobd commented:
Hi @TeofilC the GitHub Actions workflow requires approval.
